--- a/scripts/repo-add.sh.in	2023-06-06 23:55:27.077340835 +0800
+++ b/scripts/repo-add.sh.in	2023-06-06 23:53:14.868339909 +0800
@@ -274,7 +274,7 @@
 		pgpsig=$(base64 "$pkgfile.sig" | tr -d '\n')
 	fi
 
-	csize=$(wc -c < "$pkgfile")
+	csize=$(wc -c < "$pkgfile" | xargs)
 
 	# compute checksums
 	msg2 "$(gettext "Computing checksums...")"
--- a/scripts/libmakepkg/util/dirsize.sh.in	2023-06-06 23:56:47.242601380 +0800
+++ b/scripts/libmakepkg/util/dirsize.sh.in	2023-06-06 23:56:52.792603452 +0800
@@ -37,5 +37,5 @@
 				cat "$file"
 			fi
 		done < <(find . -type f -links +1 -exec @INODECMD@ {} + 2>/dev/null)
-	} | wc -c
+	} | wc -c | xargs
 }
