--- a/lib/libalpm/util.c	2023-06-03 01:22:03.464763145 +0800
+++ b/lib/libalpm/util.c	2023-06-03 01:23:12.332191398 +0800
@@ -674,10 +674,14 @@
 		}
 
 		/* use fprintf instead of _alpm_log to send output through the parent */
 		/* don't chroot() to "/": this allows running with less caps when the
 		 * caller puts us in the right root */
-		if(strcmp(handle->root, "/") != 0 && chroot(handle->root) != 0) {
-			fprintf(stderr, _("could not change the root directory (%s)\n"), strerror(errno));
+
+		// note: don't actually chroot, because "hardened runtime" fuckery means that any script
+		// executing in the chroot environment will just fucking die cryptically
+		if(strcmp(handle->root, "/") != 0) {
+			fprintf(stderr, "tried to chroot to unsupported directory '%s'\n", handle->root);
 			exit(1);
 		}
+
 		if(chdir("/") != 0) {
 			fprintf(stderr, _("could not change directory to %s (%s)\n"),
 					"/", strerror(errno));
 
