--- a/numpy/f2py/cfuncs.py	2023-05-16 13:11:30.727577474 +0800
+++ b/numpy/f2py/cfuncs.py	2023-05-16 13:11:39.089131855 +0800
@@ -548,7 +548,7 @@
       && (__STDC_VERSION__ >= 201112L) \\
       && !defined(__STDC_NO_THREADS__) \\
       && (!defined(__GLIBC__) || __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ > 12)) \\
-      && !defined(NPY_OS_OPENBSD) && !defined(NPY_OS_HAIKU)
+      && !defined(NPY_OS_OPENBSD) && !defined(NPY_OS_HAIKU) && !defined(NPY_OS_DARWIN)
 /* __STDC_NO_THREADS__ was first defined in a maintenance release of glibc 2.12,
    see https://lists.gnu.org/archive/html/commit-hurd/2012-07/msg00180.html,
    so `!defined(__STDC_NO_THREADS__)` may give false positive for the existence
