--- a/scripts/fakeroot.in	2024-03-02 00:00:20.854143320 -0500
+++ b/scripts/fakeroot.in	2024-03-02 00:00:48.818235681 -0500
@@ -35,7 +35,7 @@
 
 USEABSLIBPATH=@LDPRELOADABS@
 FAKEROOT_LIB=lib@fakeroot_transformed@@DLSUFFIX@
-PATHS=@libdir@:${FAKEROOT_PREFIX}/lib64/libfakeroot:${FAKEROOT_PREFIX}/lib32/libfakeroot
+PATHS=@libdir@:${FAKEROOT_PREFIX}/lib/libfakeroot
 FAKED=${FAKEROOT_BINDIR}/@faked_transformed@
 
 FAKED_MODE="unknown-is-root"
@@ -45,9 +45,9 @@
 
 GETOPTTEST=`getopt -T`
 if test "$?" -eq 4; then # GNU getopt
-    FAKE_TEMP=`getopt -l lib: -l faked: -l unknown-is-real -l fd-base: -l version -l help -- +l:f:i:s:ub:vh "$@"`
+    FAKE_TEMP=`getopt -l lib: -l faked: -l unknown-is-real -l fd-base: -l version -l help -- +l:f:i:s:ub:vh "${@@Q}"`
 else
-    FAKE_TEMP=`getopt l:f:i:s:ub:vh "$@"`
+    FAKE_TEMP=`getopt l:f:i:s:ub:vh "${@@Q}"`
 fi
 
 if test "$?" -ne 0; then
@@ -104,7 +104,9 @@
   shift
 done
 
-shift #get rid of the '--'
+while test "X$1" = "X--"; do
+    shift
+done
 
 # make sure the preload is available
 ABSLIB=""
