--- a/PyObjCTest/test_array_interface.py	2023-05-17 21:59:11.707194473 +0800
+++ b/PyObjCTest/test_array_interface.py	2023-05-17 21:59:16.698917063 +0800
@@ -249,7 +249,6 @@
         v = self.type2test([1, 2, 0, 3, 4])
         self.assertEqual(u, v)
 
-    @expectedFailure
     def test_pyobjc_pop(self):
         u = self.type2test([1, 2, 3, 4])
         with self.assertRaises(IndexError):
--- a/PyObjCTest/test_methodaccess.py	2023-05-17 22:05:32.246815202 +0800
+++ b/PyObjCTest/test_methodaccess.py	2023-05-17 22:05:45.523685377 +0800
@@ -239,7 +239,6 @@
         ):
             NSObject.new().alloc
 
-    @expectedFailure
     def testClassThroughInstance2(self):
         # Class methods are not accessible through instances.
         with self.assertRaisesRegex(
@@ -247,7 +246,6 @@
         ):
             Py_FirstGenSubClas.new().alloc
 
-    @expectedFailure
     def testPythonClassThroughInstance(self):
         # same as testClassThroughInstance, but without involving
         # PyObjC. This test uses properties instead of methods
