--- a/clang/lib/Driver/ToolChains/Darwin.cpp	2023-09-07 17:13:04.336268226 -0400
+++ b/clang/lib/Driver/ToolChains/Darwin.cpp	2023-09-07 18:15:32.758025277 -0400
@@ -250,6 +250,14 @@
     }
   }
 
+  // add our vendor sysroot libs
+  {
+    llvm::SmallString<128> libDir = llvm::StringRef(D.getInstalledDir()); // <install>/bin
+    llvm::sys::path::append(libDir, "..", "lib");
+    CmdArgs.push_back("-L");
+    CmdArgs.push_back(Args.MakeArgString(libDir));
+  }
+
   // Use -lto_library option to specify the libLTO.dylib path. Try to find
   // it in clang installed libraries. ld64 will only look at this argument
   // when it actually uses LTO, so libLTO.dylib only needs to exist at link
@@ -2351,6 +2359,11 @@
       addExternCSystemInclude(DriverArgs, CC1Args, Prefix + dir);
     }
   } else {
+    // and <installed_dir>/../include
+    SmallString<128> P1(D.getInstalledDir());
+    llvm::sys::path::append(P1, "..", "include");
+    addExternCSystemInclude(DriverArgs, CC1Args, P1.str());
+
     // Otherwise, add <sysroot>/usr/include.
     SmallString<128> P(Sysroot);
     llvm::sys::path::append(P, "usr", "include");
