--- a/pifpaf/drivers/redis.py	2023-05-14 21:19:59.725108429 +0800
+++ b/pifpaf/drivers/redis.py	2023-05-14 21:20:14.667295867 +0800
@@ -50,7 +50,7 @@
         super(RedisDriver, self)._setUp()
         c, _ = self._exec(
             ["redis-server", "-"],
-            stdin=("dir %s\nport %d\n"
+            stdin=("dir %s\nport %d\nenable-debug-command local\n"
                    % (self.tempdir, self.port)).encode('ascii'),
             wait_for_line="eady to accept connections")
 
@@ -59,7 +59,7 @@
             with open(cfg, "w") as f:
                 f.write("""dir %s
 port %d
-sentinel monitor pifpaf localhost %d 1"""
+sentinel monitor pifpaf 127.0.0.1 %d 1"""
                         % (self.tempdir, self.sentinel_port, self.port))
 
             c, _ = self._exec(
