--- a/third_party/gloo/cmake/Dependencies.cmake	2023-05-28 21:29:16.095118191 +0800
+++ b/third_party/gloo/cmake/Dependencies.cmake	2023-05-28 21:29:24.579838525 +0800
@@ -91,8 +91,8 @@
 
     add_library(uv_a INTERFACE IMPORTED)
     set_target_properties(uv_a PROPERTIES
-      INTERFACE_INCLUDE_DIRECTORIES ${libuv_INCLUDE_DIRS}
-      INTERFACE_LINK_LIBRARIES ${libuv_LIBRARY}
+      INTERFACE_INCLUDE_DIRECTORIES "${libuv_INCLUDE_DIRS}"
+      INTERFACE_LINK_LIBRARIES "${libuv_LIBRARY}"
       )
   endif()
 endif()
