--- a/tests/test-execute-main.c	2023-04-29 21:20:44
+++ b/tests/test-execute-main.c	2023-04-29 21:21:29
@@ -407,26 +407,26 @@
         ASSERT (fclose (fp) == 0);
 
         int fd_in = open (BASE ".tmp", O_RDONLY);
-        ASSERT (fd_in >= 0 && fd_in < 10);
+        ASSERT (fd_in >= 0 && fd_in < 11);
 
         int fd_out = open (BASE ".tmp", O_WRONLY | O_APPEND);
-        ASSERT (fd_out >= 0 && fd_out < 10);
+        ASSERT (fd_out >= 0 && fd_out < 11);
 
-        ASSERT (dup2 (fd_in, 10) >= 0);
+        ASSERT (dup2 (fd_in, 11) >= 0);
         close (fd_in);
-        fd_in = 10;
+        fd_in = 11;
 
-        ASSERT (dup2 (fd_out, 11) >= 0);
+        ASSERT (dup2 (fd_out, 12) >= 0);
         close (fd_out);
-        fd_out = 11;
+        fd_out = 12;
 
         const char *prog_argv[3] = { prog_path, "19", NULL };
         int ret = execute (progname, prog_argv[0], prog_argv, NULL,
                            false, false, false, false, true, false, NULL);
         #if defined _WIN32 && ! defined __CYGWIN__
-        ASSERT (ret == 4 + 2 * (_isatty (10) != 0) + (_isatty (11) != 0));
+        ASSERT (ret == 4 + 2 * (_isatty (11) != 0) + (_isatty (12) != 0));
         #else
-        ASSERT (ret == 4 + 2 * (isatty (10) != 0) + (isatty (11) != 0));
+        ASSERT (ret == 4 + 2 * (isatty (11) != 0) + (isatty (12) != 0));
         #endif
 
         close (fd_in);
