--- a/c/_cffi_backend.c	2023-05-06 15:21:26.564266405 +0800
+++ b/c/_cffi_backend.c	2023-05-06 15:21:33.753899707 +0800
@@ -96,7 +96,7 @@
 # define CFFI_CHECK_FFI_PREP_CIF_VAR 0
 # define CFFI_CHECK_FFI_PREP_CIF_VAR_MAYBE 0
 
-#elif defined(__APPLE__) && defined(FFI_AVAILABLE_APPLE)
+#elif defined(__APPLE__)
 
 # define CFFI_CHECK_FFI_CLOSURE_ALLOC __builtin_available(macos 10.15, ios 13, watchos 6, tvos 13, *)
 # define CFFI_CHECK_FFI_CLOSURE_ALLOC_MAYBE 1
