--- a/tests/ssl-test.c	2024-02-11 02:51:25.900233091 -0500
+++ b/tests/ssl-test.c	2024-02-11 02:51:18.331664914 -0500
@@ -735,7 +735,7 @@
 	test_init (argc, argv, NULL);
 
 #if HAVE_GNUTLS
-        char *module_path = soup_test_build_filename_abs (G_TEST_BUILT, "mock-pkcs11.so", NULL);
+        char *module_path = soup_test_build_filename_abs (G_TEST_BUILT, "mock-pkcs11.dylib", NULL);
         g_assert_true (g_file_test (module_path, G_FILE_TEST_EXISTS));
 
         g_assert (gnutls_pkcs11_init (GNUTLS_PKCS11_FLAG_MANUAL, NULL) == GNUTLS_E_SUCCESS);
